### DISCUSSION TITLE
Permission Denied error more verbose?

### DISCUSSION THREAD

**Discussion by paulboudewijn** [directus/directus#9972] [Feature Requests]
A user with limited permissions gets a "permission denied" error when editing a junction table through the m2m interface. 
The admin user doesn't get this error so it's definitely a configuration issue. 
But since the "permission denied" error doesn't indicate which object it applies to, it's quite a challenge to resolve.

Could the object name be added to the "permission denied" error message, or is that a security risk?

**Comment by rijkvanzanten** [comment]
> Could the object name be added to the "permission denied" error message, or is that a security risk?

It depends on what the request was that was being made 🤔 If you request `/items/example/1`, we could return the values `example:1`, as you've already explicitly requested them. However, we can't return `authors:rijk` when fetching `/articles?fields=*.*.*`, as that would leak the items that exist in the system. 

> The admin user doesn't get this error so it's definitely a configuration issue.

Admin users get a stack trace and some other private info in error messages, this is by design 👍🏻

**Comment by paulboudewijn** [comment]
What I meant was, I would like to know which collection field causes the permission denied error. So if the user isn't allowed to edit the `example.author` field, I would like the error message to say `[FORBIDDEN] You don't have permission to edit example authors.` instead of:

![image](https://user-images.githubusercontent.com/17253180/142889443-7bbf171a-9cb6-4933-8676-d10d0449f35a.png)

You would think that a field that a user is not allowed to change would be read-only in the app, but then this error would not occur. Right?

**Reply by rijkvanzanten** [reply]
Yeah that makes sense

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
